hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/addressbook/NodeCreateSuite.java

```diff
@@ -17,41 +17,344 @@
+                        .setPort(123)
+                        .build());
+        return hapiTest(nodeCreate("nodeCreate")
+                .serviceEndpoint(serviceEndpoints)
+                .hasKnownStatus(SERVICE_ENDPOINTS_EXCEEDED_LIMIT));
+    }
 
+    /**
+     * Check that node creation succeeds with gossip and service endpoints using domain names and all optional fields are recorded.
+     * @see <a href="https://github.com/hashgraph/hedera-improvement-proposal/blob/main/HIP/hip-869.md#specification">HIP-869</a>
+     */
+    @HapiTest
+    @Tag(EMBEDDED)
+    final Stream<DynamicTest> allFieldsSetHappyCaseForDomains() {
         return hapiTest(
-                nodeCreate("ntb").description(description),
-                viewNode(
-                        "ntb", node -> assertEquals(description, node.description(), "Node was created successfully")));
+                newKeyNamed(ED_25519_KEY).shape(KeyShape.ED25519),
+                overriding("nodes.gossipFqdnRestricted", "false"),
+                nodeCreate("nodeCreate")
+                        .description("hello")
+                        .gossipCaCertificate("gossip".getBytes())
+                        .grpcCertificateHash("hash".getBytes())
+                        .accountId(asAccount("0.0.100"))
+                        .gossipEndpoint(GOSSIP_ENDPOINTS)
+                        .serviceEndpoint(SERVICES_ENDPOINTS)
+                        .adminKeyName(ED_25519_KEY)
+                        .hasPrecheck(OK)
+                        .hasKnownStatus(SUCCESS),
+                viewNode("nodeCreate", node -> {
+                    assertEquals("hello", node.description(), "Description invalid");
+                    assertEquals(
+                            ByteString.copyFrom("gossip".getBytes()),
+                            ByteString.copyFrom(node.gossipCaCertificate().toByteArray()),
+                            "Gossip CA invalid");
+                    assertEquals(
+                            ByteString.copyFrom("hash".getBytes()),
+                            ByteString.copyFrom(node.grpcCertificateHash().toByteArray()),
+                            "GRPC hash invalid");
+                    assertEquals(100, node.accountId().accountNum(), "Account ID invalid");
+                    assertEqualServiceEndpoints(GOSSIP_ENDPOINTS, node.gossipEndpoint());
+                    assertEqualServiceEndpoints(SERVICES_ENDPOINTS, node.serviceEndpoint());
+                    assertNotNull(node.adminKey(), "Admin key invalid");
+                }));
     }
 
+    /**
+     * Check that node creation succeeds with gossip and service endpoints using IPs and all optional fields are recorded.
+     * @see <a href="https://github.com/hashgraph/hedera-improvement-proposal/blob/main/HIP/hip-869.md#specification">HIP-869</a>
+     */
+    @HapiTest
+    @Tag(EMBEDDED)
+    final Stream<DynamicTest> allFieldsSetHappyCaseForIps() {
+
+        return hapiTest(
```

Comment:
All these tests are signing with genesis key, we can also change one test that signs with addressBookAdmin. Even though on genesis they both use same key, just good to validate